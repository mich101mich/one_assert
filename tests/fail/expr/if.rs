fn main() {
    let x = 1;
    one_assert::assert!(if x == 1 { 1 } else { 2 });            // not bool
    one_assert::assert!(if x == 1 { 1 } else { 2 } else { 3 }); // too many else
    one_assert::assert!(if x == 1 { true });                    // no else
    one_assert::assert!(if x == 1 { true } else { 1 });         // mismatched types
    one_assert::assert!(if x == 1 { true } else if true { false } else if false { true } ); // no final else
    one_assert::assert!(if x == 1 { true } else if true { false } else while false {} );    // else while instead of else if
}
