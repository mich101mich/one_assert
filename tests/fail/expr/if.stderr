error: condition has to be followed by a comma, if a message is provided
 --> tests/fail/expr/if.rs:4:52
  |
4 |     one_assert::assert!(if x == 1 { 1 } else { 2 } else { 3 }); // too many else
  |                                                    ^^^^

error: expected `if` or curly braces
 --> tests/fail/expr/if.rs:8:72
  |
8 |     one_assert::assert!(if x == 1 { true } else if true { false } else while false {} );    // else while instead of else if
  |                                                                        ^^^^^

error[E0308]: mismatched types
 --> tests/fail/expr/if.rs:3:37
  |
3 |     one_assert::assert!(if x == 1 { 1 } else { 2 });            // not bool
  |                                     ^ expected `bool`, found integer

error[E0308]: mismatched types
 --> tests/fail/expr/if.rs:3:48
  |
3 |     one_assert::assert!(if x == 1 { 1 } else { 2 });            // not bool
  |                                                ^ expected `bool`, found integer

error[E0317]: `if` may be missing an `else` clause
 --> tests/fail/expr/if.rs:5:25
  |
5 |     one_assert::assert!(if x == 1 { true });                    // no else
  |                         ^^^^^^^^^^^^----^^
  |                         |           |
  |                         |           found here
  |                         expected `bool`, found `()`
  |
  = note: `if` expressions without `else` evaluate to `()`
  = help: consider adding an `else` block that evaluates to the expected type

error[E0308]: mismatched types
 --> tests/fail/expr/if.rs:6:51
  |
6 |     one_assert::assert!(if x == 1 { true } else { 1 });         // mismatched types
  |                                                   ^ expected `bool`, found integer

error[E0317]: `if` may be missing an `else` clause
 --> tests/fail/expr/if.rs:7:72
  |
7 |     one_assert::assert!(if x == 1 { true } else if true { false } else if false { true } ); // no final else
  |                                                                        ^^^^^^^^^^^----^^
  |                                                                        |          |
  |                                                                        |          found here
  |                                                                        expected `bool`, found `()`
  |
  = note: `if` expressions without `else` evaluate to `()`
  = help: consider adding an `else` block that evaluates to the expected type
